src/web/web.cpp

```diff
@@ -0,0 +1,459 @@
+// Copyright (c) rAthena Dev Teams - Licensed under GNU GPL
+// For more information, see LICENCE in the main folder
+
+#include "web.hpp"
+
+#include <chrono>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+#include <thread>
+
+#include "../common/cli.hpp"
+#include "../common/core.hpp"
+#include "../common/malloc.hpp"
+#include "../common/md5calc.hpp"
+#include "../common/mmo.hpp"
+#include "../common/msg_conf.hpp"
+#include "../common/random.hpp"
+#include "../common/showmsg.hpp"
+#include "../common/socket.hpp" //ip2str
+#include "../common/sql.hpp"
+#include "../common/strlib.hpp"
+#include "../common/timer.hpp"
+#include "../common/utilities.hpp"
+#include "../common/utils.hpp"
+#include "../config/core.hpp"
+
+#include "charconfig_controller.hpp"
+#include "emblem_controller.hpp"
+#include "http.hpp"
+#include "userconfig_controller.hpp"
+#include "merchantstore_controller.hpp"
+
+
+using namespace rathena;
+
+#define WEB_MAX_MSG 30				/// Max number predefined in msg_conf
+static char* msg_table[WEB_MAX_MSG];	/// Web Server messages_conf
+
+struct Web_Config web_config {};
+struct Inter_Config inter_config {};
+std::shared_ptr<httplib::Server> http_server;
+
+int login_server_port = 3306;
+char login_server_ip[64] = "127.0.0.1";
```

Comment:
Would it be ok to convert these to string like what we did for #7006? Technically the limit will be then capped to 1024 from `inter_config_read` but that should be large enough for anyone's use for now.